packages/angular_devkit/build_angular/src/browser/tests/options/watch_spec.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { mergeMap, take, tap, timeout } from 'rxjs/operators';
+import { buildWebpackBrowser } from '../../index';
+import { BASE_OPTIONS, BROWSER_BUILDER_INFO, describeBuilder } from '../setup';
+
+describeBuilder(buildWebpackBrowser, BROWSER_BUILDER_INFO, (harness) => {
+  describe('Option: "watch"', () => {
+    it('does not wait for file changes when false', (done) => {
+      harness.useTarget('build', {
+        ...BASE_OPTIONS,
+        watch: false,
+      });
+
+      // If the build waits then it will timeout with a custom 30 second timeout
+      let count = 0;
+      harness
+        .execute()
+        .pipe(timeout(30000))
+        .subscribe({
+          complete() {
+            expect(count).toBe(1);
+            done();
+          },
+          next({ result }) {
+            count++;
+            expect(result?.success).toBe(true);
+          },
+          error(error) {
+            done.fail(error);
+          },
+        });
+    });
+
+    it('watches for changes to the main file', async () => {
+      harness.useTarget('build', {
+        ...BASE_OPTIONS,
+        main: 'src/main.ts',
+        watch: true,
+      });
+
+      let count = 0;
+      await harness
+        .execute()
+        .pipe(
+          timeout(30000),
```

Comment:
Do we still need the timeout?

Reply:
It helps in error cases such as watch not picking up the change.  With this timeout being less than the jasmine timeout, the observable will be guaranteed to be closed out.  Whereas with just the Jasmine timeout, the watch could trigger after the test fails and still execute the operators.  I was actually considering adding an automatic builder abort into the jasmine test harness to mitigate this without the extra timeouts.

Reply:
>  I was actually considering adding an automatic builder abort into the jasmine test harness to mitigate this without the extra timeouts.

I do like that, although that can be done in a seperate PR.